File: app/llm/mock_adapter.py
"""Mock 适配器，用于测试（支持流式）。"""

from __future__ import annotations

from typing import Iterator

from app.llm.base import (
    BaseAdapter
)
from app.llm.types import LLMRequest, LLMResponse, LLMUsage, ParsedResponse, StreamChunk, TextBlock


class MockAdapter(BaseAdapter):
    """Mock LLM 适配器，支持自定义响应文本。"""

    def __init__(self, response_text: str = 'mock') -> None:
        self._response_text = response_text

    def complete(self, req: LLMRequest) -> LLMResponse:
        _ = req
        return LLMResponse(
            text=self._response_text,
            usage=LLMUsage(prompt_tokens=10, completion_tokens=5, total_tokens=15),
        )

    def parse_response(self, response: LLMResponse) -> ParsedResponse:
        text = response.text
        return ParsedResponse(
            text=text,
            blocks=[TextBlock(type='text', text=text)],
            tool_calls=[],
            raw=response.raw,
            usage=response.usage,
        )

    def stream(self, req: LLMRequest) -> Iterator[StreamChunk]:
        """逐词模拟流式输出。"""
        _ = req
        words = self._response_text.split()
        for i, word in enumerate(words):
            delta = word if i == 0 else f' {word}'
            yield StreamChunk(text_delta=delta)
        yield StreamChunk(
            is_done=True,
            usage=LLMUsage(prompt_tokens=10, completion_tokens=len(words), total_tokens=10 + len(words)),
        )

File: miniagents-assembled/src/main/java/com/codex/miniagents/llm/adapter/MockAdapter.java
package com.codex.miniagents.llm.adapter;

import com.codex.miniagents.llm.model.LlmContentBlock;
import com.codex.miniagents.llm.model.LlmRequest;
import com.codex.miniagents.llm.model.LlmResponse;
import com.codex.miniagents.llm.model.LlmUsage;
import com.codex.miniagents.llm.model.ParsedResponse;
import com.codex.miniagents.llm.model.StreamChunk;
import com.codex.miniagents.llm.model.TextBlock;

import java.util.Iterator;
import java.util.List;
import java.util.HashMap;
import java.util.ArrayList;

public class MockAdapter extends LlmAdapter {
    private final String responseText;

    public MockAdapter() {
        this("mock");
    }

    public MockAdapter(String responseText) {
        this.responseText = responseText;
    }

    @Override
    public LlmResponse complete(LlmRequest request) {
        return LlmResponse.builder()
            .text(responseText)
            .raw(new HashMap<>())
            .usage(LlmUsage.builder().promptTokens(10).completionTokens(5).totalTokens(15).build())
            .build();
    }

    @Override
    public ParsedResponse parseResponse(LlmResponse response) {
        List<LlmContentBlock> blocks = new ArrayList<>();
        blocks.add(new TextBlock("text", response.getText()));
        return ParsedResponse.builder()
            .text(response.getText())
            .blocks(blocks)
            .toolCalls(new ArrayList<>())
            .raw(response.getRaw())
            .usage(response.getUsage())
            .build();
    }

    @Override
    public Iterator<StreamChunk> stream(LlmRequest req) {
        String[] words = responseText.isEmpty() ? new String[0] : responseText.split("\\s+");
        return new Iterator<>() {
            private int index = 0;

            @Override
            public boolean hasNext() {
                return index <= words.length;
            }

            @Override
            public StreamChunk next() {
                if (!hasNext()) {
                    throw new java.util.NoSuchElementException();
                }
                if (index == words.length) {
                    index++;
                    return StreamChunk.builder()
                        .done(true)
                        .usage(LlmUsage.builder()
                            .promptTokens(10)
                            .completionTokens(words.length)
                            .totalTokens(10 + words.length)
                            .build())
                        .build();
                }
                String delta = index == 0 ? words[index] : " " + words[index];
                index++;
                return StreamChunk.builder().textDelta(delta).build();
            }
        };
    }
}
